pallets/pallet-deposit-storage/src/fungible/tests/inspect_hold.rs

```diff
@@ -0,0 +1,60 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use frame_support::traits::fungible::{InspectHold, MutateHold};
+use sp_runtime::{traits::Zero, AccountId32};
+
+use crate::{
+	fungible::tests::mock::{DepositNamespace, ExtBuilder, TestRuntime, OTHER_ACCOUNT, OWNER},
+	Pallet, PalletDepositStorageReason,
+};
+
+#[test]
+fn balance_on_hold() {
```

Comment:
can we also maybe add a test for creating a second hold with a different `HoldReason`? I would expect two SystemDeposits accumulated into the one holdReason in the Currency pallet. You prob have to change the mock setup

Reply:
But why would you test the `Currency` pallet interactions?

Reply:
Actually, that is possible. The deposit is storing the account.

Reply:
as a try_runtime test

Reply:
You helped me identify a very big bug. I addressed it in here: https://github.com/KILTprotocol/kilt-node/pull/823/commits/6f3e1ace0d705d8cedc9a5b12e97a5b4e0d2e641.

Simply we can't forward the `set_balance_on_hold` to the underlying currency directly, as that would override the balance for the overarching `HoldReason::Deposit`. What we need to do (I've done it in the commit), is to call either `increase_balance_on_hold` or `decrease_balance_on_hold` depending on weather we are reserving or releasing holds.

This was a very very good catch that I did not write a test for before, but now I did.

Reply:
Hey, I opened a new PR on top of this where I add `try-state` checks. Please check that out, so we can merge that on top of this before merging this. Now I feel much more confident everything works as expected, also thanks to you 😀

https://github.com/KILTprotocol/kilt-node/pull/830